blueprint/core/blueprint_ReactApplicationRoot.cpp

```diff
@@ -9,7 +9,196 @@
 
 #pragma once
 
+#include "blueprint_ReactApplicationRoot.h"
+
 
 namespace blueprint
 {
+
+    ReactApplicationRoot::ReactApplicationRoot(std::shared_ptr<EcmascriptEngine> ee)
+        : viewManager(this)
+        , engine(ee)
+    {
+        JUCE_ASSERT_MESSAGE_THREAD
+        bindNativeRenderingHooks();
+    }
+
+    ReactApplicationRoot::ReactApplicationRoot()
+        : ReactApplicationRoot(std::make_shared<EcmascriptEngine>())
+    {
+
+#if JUCE_DEBUG
+        // Enable keyboardFocus to support CTRL-D/CMD-D debug attachment.
+        setWantsKeyboardFocus(true);
```

Comment:
Should this not be moved up into the constructor above? i.e. enable the debug key bindings in both use cases of `ReactApplicationRoot` ?

Reply:
Err I need to update the comments everywhere. I think ReactAppRoot by default will have no hot-reloading, but the thought is that if you want it, just make yourself a little `HotReloadingHarness` and kick it off.

And +1 to moving that up to the previous constructor, good catch